resources/type-system/subtyping.md

```diff
@@ -242,6 +242,140 @@ that the choice of common variable names avoid capture.  It is valid to choose
 the `Xi` or the `Yi` for `Zi` so long as capture is avoided*
 
 
+## Algorithmic subtyping: version 2
+
+This version of the algorithmic rules re-orders the rules to move all of the
+rules relevant to non-nullability as early as possible.  It also simplifies the
+algorithm somewhat by allowing specific rules to "fall through" to later rules
+when it is clearer to do so.  This presentation is intended to be equivalent to
+the above.  It is motivated by pragmatic concerns from the implementations.
+
+### Rules
+
+We say that a type `T` is **nullable** if `Null <: T`.  This is equivalent to
+the syntactic criterion that `T` is any of:
+  - `Null`
+  - `S?` for some `S`
+  - `S*` for some `S` where `S` is nullable
```

Comment:
note that we don't have to cover this case ourselves in the analyzer; we definite `[S/S'?]S* = S'?`.